src/mount.rs

```diff
@@ -0,0 +1,655 @@
+use lru::LruCache;
+use rayon::iter::{IntoParallelIterator, ParallelIterator};
+use std::{
+    borrow::Cow,
+    cmp::Ordering,
+    ffi::OsStr,
+    fs,
+    io::{self, ErrorKind, Read},
+    iter::Peekable,
+    num::NonZeroUsize,
+    ops::ControlFlow,
+    path::{Component, Path, PathBuf},
+    sync::{Arc, Mutex},
+    time::Duration,
+};
+use tracing::{error, info, trace};
```

Comment:
nit: please order imports as paragraphs of

1. std
2. dependencies
3. this crate